### ISSUE TITLE
[webpack] Error: Entry module not found: Error: Can't resolve 'buble' in

### ISSUE THREAD

**Issue by nhowell** [laravel/elixir#643]
I upgraded to Laravel 5.3 and I'm trying to make use of `mix.webpack('app.js')` but I'm getting the following error when running `gulp`:
```
Error: Entry module not found: Error: Can't resolve 'buble' in ...
```

I tried deleting "./node_modules" and did a `npm install` with a clean slate, too. And I confirmed that the "buble" package is present in "./node_modules/laravel-elixir-webpack-official/node_modules/buble".

I'm running:
Node v4.5.0 (current LTS release)
NPM 2.15.9
And relevant part of package.json:
```
"devDependencies": {
    "gulp": "^3.9.1",
    "laravel-elixir": "^6.0.0-9",
    "laravel-elixir-vue": "^0.1.7",
    "laravel-elixir-webpack-official": "^1.0.5",
    "vue": "^1.0.26"
}
```

**Comment by tankerkiller125** [comment]
I found a fix for this by doing `npm install buble-loader`

**Comment by pakogn** [comment]
it's better to update your node version to 6 or above and the problem will be solved.

**Comment by nhowell** [comment]
> it's better to update your node version to 6 or above and the problem will be solved.

And when that's not an option? I don't think it's unreasonable to expect it to work on the LTS release of Node. Enterprises usually prefer the LTS releases.

**Comment by nhowell** [comment]
Also, from what I can tell, there's no explicit requirement for any minimum version of Node documented anywhere in this project.

**Comment by pakogn** [comment]
I agree with you in all points that you're commenting. I think that if Laravel provides an initial stuff to work with, it should work fine out of the box.

Also I was noticing that the node download link that is included in the documentation points to => https://nodejs.org/en/download/ which offers the 4.5 lts version.

But in homestead, it uses the 6 version. https://github.com/laravel/settler/blob/master/scripts/provision.sh#L35(https://github.com/laravel/settler/blob/master/scripts/provision.sh#L35)

May be they obviate that you are working with the latest relase of the stuff.

Have you ever tried this with node 6? it would be interesting to watch that this works out of the box and  the packages remains the same, then know what is missing in 4.5 and fix the detail.

**Comment by tankerkiller125** [comment]
I'm currently using node version 6 and I still get the buble error. Along with other problems like add-module-export unknown plugin.

**Comment by pakogn** [comment]
Kind of weird, I had the same issue and after wasting a lot of time, I tested with node v6 and all went OK with Laravel 5.3 out of the box and with my web pack.  I'll check that tomorrow.

**Comment by victorhguillen** [comment]
I use node v6 it work out the box.

**Comment by tankerkiller125** [comment]
It seems that if you have the laravel installed already and then you upgrade to node v6 then npm install will still fail. However if you create a new project and you already have node v6 then it won't. Very weird bug.

**Comment by pakogn** [comment]
Yes,  kind of weird, I'm using vuex so I need to manage components and some other stuff. Then I found convenient to use babel instead of buble and with some upgrades now my workaround doesn't work. I spent some minutes to got it work again. Finally the problem was the package versions.

This is not the first time that I'm in this scenario so I one thing that I would recommend is that, once you have it working as spected make a npm shrinkwrap, turn the versions statically so the next time you have to install all modules you will have the correct versions and the same workaround to develop.

**Comment by luchaos** [comment]
For me, it turned out to be the enabled **legacy bundling** that one of the many javascript ecosystem thingies can't deal with. I upgraded to node v6, again to v6.6.0 via n, npm is at 3.10 now, but it just would not work. I disabled legacy-bundling and suddenly everything works as expected.

[The stackoverflow post where I endorse using legacy-bundling, which, to my sorrow, does not seem to be a good idea after all.](http://stackoverflow.com/a/35227212/580651)

**Comment by unitedworx** [comment]
It appears that the bubble error occurs with node LTS version v4.6.0 however it works fine with current node version v6.7.0.

This was resolved here by deletetign node_modules, upgrading to the latest node version (v6.7.0) running "npm install" and then "gulp"

**Comment by Ingosmar89219** [comment]
the tankerkiller125 's Solution Works for me :D thanks a lot guy

**Comment by MarkGlenn07** [comment]
@tankerkiller125 thank you man. it works. helps me save some time
